ttnn/cpp/ttnn/operations/experimental/reshape/reshape_pybind.cpp

```diff
@@ -0,0 +1,72 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "reshape_pybind.hpp"
+#include "reshape.hpp"
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "ttnn/cpp/pybind11/decorators.hpp"
+
+#include "ttnn/types.hpp"
+
+#include "ttnn/tensor/tensor.hpp"
+#include "ttnn/tensor/tensor_impl.hpp"
+
+
+namespace ttnn::operations::experimental::reshape::detail {
+namespace py = pybind11;
+
+void py_bind_reshape(py::module& module) {
+    auto doc = R"doc(
+
+        Note: for a 0 cost view, the following conditions must be met:
+            * the last dimension must not change
+            * In Tiled the second last two dimensions must not change OR there is no padding on the second last dimension
+
+        Args:
+            * input_tensor: Input Tensor.
+            * new_shape: New shape of tensor.
+
+        Returns:
+            ttnn.Tensor: the output tensor with the new shape.
+
+        Example:
+
+            >>> tensor = ttnn.from_torch(torch.tensor((1, 4), dtype=torch.bfloat16), device=device)
```

Comment:
This example is actually unsafe for an experimental.reshape and should not be what we give as an example. change it to maybe (2,1,4) input to (2,1,1,4) output for instance?